### ISSUE TITLE
Custom regex-based rule always highlights line 1, column 1

### ISSUE THREAD

**Issue by niksawtschuk** [realm/SwiftLint#2513]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I am creating a custom regex-based rule which checks for things that are marked as `@objc` and `public` which are not properly prefixed using our own guideline (`SBLClassName`). We do this to make sure we don't cause class name conflicts with apps that our SDK lives inside. It seems like the regex itself is working (it properly fails on files which contain violations), but the line/column number given in the output is always line 1, column 1. This isn't very helpful and doesn't seem to be happening for the default rules.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths
Linting 'test.swift' (1/1)
/Users/nik/tmp/swiftlint/test.swift:1:1: error: Public entities exposed to Objective-C must be prefixed Violation: Expose public entities using @objc(SBLClassName). (objc_public_prefixed)
Done linting! Found 1 violation, 1 serious in 1 file.
```

### Environment

* SwiftLint version: **0.29.1**
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
whitelist_rules:
- custom_rules

custom_rules:
  objc_public_prefixed:
    name: "Public entities exposed to Objective-C must be prefixed"
    message: "Expose public entities using @objc(SBLClassName)."
    severity: error
    regex: '^(.*public)?.*@objc(Members)?(?!\(SBL.*\))(.*public)?.*(class|protocol|enum).*$'
```

* Xcode version: **Version 10.1 (10B61)**

Here's an example file while exhibits the behaviour in question. As you can see, the line it should be outputting is line 7.

```swift
//
// this class name should be prefixed
//

import something

@objc public class ClassName: NSObject {

}
```

**Comment by marcelofabri** [comment]
The warning will be generated on the first location that the regex matches. In this case, it seems like you could tweak the regex to remove the initial `.*`.

**Comment by niksawtschuk** [comment]
@marcelofabri you are right. I didn't realize that the character position indicated in the message would be the first character of the regex match, but I'm not sure how else it would work after thinking about it 🤣. 

I wonder if it would be valuable to print out the range of the regex match as well as the beginning for custom rule violations.
